src/storage/src/source/postgres.rs

```diff
@@ -273,55 +271,29 @@ async fn postgres_replication_loop_inner(
     // Buffer rows from snapshot to retract and retry, if initial snapshot fails.
     // Postgres sources cannot proceed without a successful snapshot.
     let mut snapshot_tx = Transaction::new();
-    loop {
-        let file =
-            tokio::fs::File::from_std(tempfile::tempfile().map_err(|e| SourceReaderError {
-                inner: SourceErrorDetails::FileIO(e.to_string()),
-            })?);
-        let mut writer = tokio::io::BufWriter::new(file);
-        match task_info
-            .produce_snapshot(&mut snapshot_tx, &mut writer)
-            .await
-        {
-            Ok(_) => {
-                info!(
-                    "replication snapshot for source {} succeeded",
-                    &task_info.source_id
-                );
-                snapshot_tx
-                    .close(task_info.lsn, &task_info.sender, &task_info.activator)
-                    .await;
-                break;
-            }
-            Err(ReplicationError::Recoverable(e)) => {
-                writer.flush().await.map_err(|e| SourceReaderError {
-                    inner: SourceErrorDetails::Initialization(e.to_string()),
-                })?;
-                warn!(
-                    "replication snapshot for source {} failed, retrying: {}",
-                    &task_info.source_id, e
-                );
-                let reader = BufReader::new(writer.into_inner().into_std().await);
-                let res = PostgresTaskInfo::revert_snapshot(&mut snapshot_tx, reader)
-                    .await
-                    .map_err(|e| SourceReaderError {
-                        inner: SourceErrorDetails::FileIO(e.to_string()),
-                    });
-                snapshot_tx
-                    .close(task_info.lsn, &task_info.sender, &task_info.activator)
-                    .await;
-
-                res?;
-            }
-            Err(ReplicationError::Fatal(e)) => {
-                return Err(SourceReaderError {
-                    inner: SourceErrorDetails::Initialization(e.to_string()),
-                })
-            }
+    match task_info.produce_snapshot(&mut snapshot_tx).await {
+        Ok(_) => {
+            info!(
+                "replication snapshot for source {} succeeded",
+                &task_info.source_id
+            );
+            snapshot_tx
+                .close(task_info.lsn, &task_info.sender, &task_info.activator)
+                .await;
         }
+        Err(ReplicationError::Recoverable(_)) => {
+            error!(
+                "replication snapshot for source {} failed",
+                &task_info.source_id
+            );
 
-        // TODO(petrosagg): implement exponential back-off
-        tokio::time::sleep(Duration::from_secs(3)).await;
+            panic!("gus");
```

Comment:
Needs a real panic message! Maybe just swap the `error!` for a `panic!`?

Reply:
Also, can you add a comment about why we're doing this? Something along the lines of: in the future we probably want to handle this more gracefully, to avoid stressing out any monitoring tools, but for now panicking is the easiest way to dump the data in the pipe.